### DISCUSSION TITLE
Cannot load assembly Microsoft.Data.SqlClient

### DISCUSSION THREAD

**Discussion by deybich** [linq2db/linq2db#3069] [Q&A] [answered]
When you create a connection to sql server 2012 using the linq2db version 3.4.0, I'm receiving the message "Cannot load assembly Microsoft.Data.SqlClient", but if I use the version 2.6.0 I can stablish connection without any issue.

Apparently when you install the version 3.4.0 the installation does not adding the Microsoft.Data.SQLClient or System.Data.SQLClient.

**Comment by MaceWindu** [comment, marked as answer]
2.6.0 works only because it doesn't support `Microsoft.Data.SQLClient` and use `System.Data.SQLClient` which is a part of .net framework.

If you want to use 3rd party provider, you should add reference to it to your project
